src/hmem_ze.c

```diff
@@ -406,7 +406,7 @@ int ze_hmem_get_shared_handle(int dev_fd, void *dev_buf, int *ze_fd,
 	int ret;
 
 	assert(dev_fd != -1);
-	ret = ze_hmem_get_handle(dev_buf, (void **) &ze_handle);
+	ret = ze_hmem_get_handle(dev_buf, 0, (void **) &ze_handle);
```

Comment:
Something is off in the abstractions.  The PR adds a new size parameter, which we ignore and pass in an invalid value here...

Reply:
I think the line of thinking was  that in ```smr_format_ipc``` there is this sequence of operations:

```
ret = ofi_hmem_get_base_addr(cmd->msg.data.ipc_info.iface, ptr, &base,
				     &cmd->msg.data.ipc_info.base_length);
...
ret = ofi_hmem_get_handle(cmd->msg.data.ipc_info.iface, base,
				   (void **)&cmd->msg.data.ipc_info.ipc_handle);
```
and since the rocr function called internally in ofi_hmem_get_handle requires the buffer size to retrieve the ipc handle for that buffer,  the code wanted to avoid having to 'rediscover' the buffer size internally in that function again.

I agree however, that if there are other locations in the code that do not follow this pattern, the interface change might look unnecessary. In addition, I am not sure that this interface change is essential for this pr, its an optimization  that could also be discussed or introduced in a separate pr.